### ISSUE TITLE
@imports inside CSS modules are ordered differently than non-modules

### ISSUE THREAD

**Issue by shepmaster** [parcel-bundler/parcel#7950]
# 🐛 bug report

## 🎛 Configuration (.babelrc, package.json, cli command)

**package.json**

```json
{
  "name": "repro",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "devDependencies": {
    "parcel": "^2.4.1"
  },
  "dependencies": {
    "normalize.css": "^8.0.1"
  }
}
```

## 🤔 Expected Behavior

I expect that normalize.css will occur in the output CSS before the custom CSS, as the `@import` comes first. This should work if I'm using CSS modules or not.

## 😯 Current Behavior

**styles.css** (expected)

```css
/* contents of normalize.css */

h1 {
  color: #ff69b4;
  font-family: cursive;
}
```

**styles.module.css** (unexpected)

```css
h1 {
  color: #ff69b4;
  font-family: cursive;
}

/* contents of normalize.css */
```

## 🔦 Context

When the CSS is in an unexpected order, I have to resort to using `!important` to override things that normally I wouldn't need to.

## 💻 Code Sample

**src/index.html**

```html
<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8"/>
    <title>My First Parcel App</title>
    <link rel="stylesheet" href="styles.css" />
  </head>
  <body>
    <h1>Hello, World!</h1>
  </body>
</html>
```

**src/styles.css** and **src/styles.module.css**

```css
@import "npm:normalize.css";

h1 {
  color: hotpink;
  font-family: cursive;
}
```

1. Run `yarn parcel src/index.html --no-cache` and view the generated stylesheet.
1. Change `<link rel="stylesheet" href="styles.css" />` to `<link rel="stylesheet" href="styles.module.css" />` 
1. Run `yarn parcel src/index.html --no-cache` and view the generated stylesheet.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.4.1
| Node             | 17.9.0
|Yarn         | 1.22.18
| Operating System | macOS 12.3.1

# 🔍 Related?

- #5469
- #5840
- #7844

**Comment by devongovett** [comment]
hmm, this is working correctly for me. One weird thing in your example is importing a CSS module from HTML rather than JS. I don't think that's really supported. If you import it from JS, do things work better?

**Comment by shepmaster** [comment]
> this is working correctly for me

Similar comments have been added to the other related issues, which makes me think that there's something different in the environments. For example, I've seen similar things where two OSes / filesystems enumerated files in a directory in a different order from each other.

For what it's worth, I've replayed the steps and still see the unexpected behavior:

```
% mkdir repro2
% cd repro2
% pbpaste > package.json
% mkdir src
% pbpaste > src/index.html
% pbpaste > src/styles.css
% cp src/styles.css src/styles.module.css
% yarn
% yarn parcel src/index.html --no-cache
% emacs src/index.html
% yarn parcel src/index.html --no-cache
```

> your example is importing a CSS module from HTML rather than JS

Yep, and I don't honestly recall why I wrote that originally. I _think_ I wanted to use `composes` inside of some global element styles. My real application doesn't currently use any module-specific features in that CSS file, so I've renamed it to work around this issue.

> I don't think that's really supported

If it's not supported, can it be prevented / warned about?
